otter/convergence/model.py

```diff
@@ -126,6 +154,69 @@ def __init__(self):
         assert self.state in ServerState.iterconstants(), \
             "%r is not a ServerState" % (self.state,)
 
+    @classmethod
+    def group_id_from_metadata(cls, metadata):
```

Comment:
I kinda feel like this should just be a toplevel function (like get_service_metadata) since it doesn't do anything with NovaServer directly